### PR TITLE
ST: Add test for connect and connects2i with same name

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

_Select the type of your PR_

- Enhancement / new feature

### Description

Add system test coverage for https://github.com/strimzi/strimzi-kafka-operator/pull/2463

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectST#testKafkaConnectorWithConnectAndConnectS2IWithSameName,ConnectS2IST#testKafkaConnectorWithConnectS2IAndConnectWithSameName

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 1]
```diff
@@ -332,6 +335,91 @@ void testJvmAndResources() {
...
     }
 
+    @Test
+    void testKafkaConnectorWithConnectS2IAndConnectWithSameName() {
+        String topicName = "test-topic-" + new Random().nextInt(Integer.MAX_VALUE);
+        String connectClusterName = "connect-cluster";
+        String connectS2IClusterName = "connect-s2i-cluster";
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3).done();
+        // Crate connect cluster with default connect image
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: You have the comments from the oposite case - here is now S2I and on line 376 is regular connect.

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 2]
```diff
@@ -332,6 +335,91 @@ void testJvmAndResources() {
...
+            .editMetadata()
+                .addToLabels("type", "kafka-connect-s2i")
+                .addToAnnotations("strimzi.io/use-connector-resources", "true")
+            .endMetadata()
+                .editSpec()
+                .addToConfig("group.id", connectClusterName)
+                .addToConfig("offset.storage.topic", connectClusterName + "-offsets")
+                .addToConfig("config.storage.topic", connectClusterName + "-config")
+                .addToConfig("status.storage.topic", connectClusterName + "-status")
+            .endSpec().done();
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: Right now both this and the connect test are in following order:
* Create ConnectS2I
* Create Connector
* Create conflicting Connect with the same name.

I wonder if the better order would be:
* Create ConnectS2I
* Create conflicting Connect with the same name.
* Create Connector

Or mybe at least try that for example connector update still works after you create the conflicting Connect.

Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: That make sense, I will update it.

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only profile=regression testcase=ConnectST#testKafkaConnectorWithConnectAndConnectS2IWithSameName,ConnectS2IST#testKafkaConnectorWithConnectS2IAndConnectWithSameName

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 3]
```diff
@@ -332,6 +335,101 @@ void testJvmAndResources() {
...
+
+        // Crate connect cluster with default connect image
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]:
```suggestion
        // Create connect cluster with default connect image
```

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java [hunk 4]
```diff
@@ -332,6 +335,101 @@ void testJvmAndResources() {
...
+                .withClassName("org.apache.kafka.connect.file.FileStreamSinkConnector")
+                .addToConfig("topics", topicName)
+                .addToConfig("file", "/tmp/test-file-sink.txt")
+                .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+            .endSpec().done();
+        KafkaConnectUtils.waitForConnectorReady(CLUSTER_NAME);
+
+        // Wait for first reconciliation
+        StUtils.waitForReconciliation(testClass, testName, NAMESPACE);
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: Reconciliation of what?

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java (hunk 2 above)
Comment (tombentley) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectS2IST.java]: You probably need to test both to ensure the first-created-resource-wins strategy.

**Pull request merged**